src/project-manager.ts

```diff
@@ -315,17 +315,13 @@ export class ProjectManager implements Disposable {
 	 * @param uri URI of the file to process
 	 * @return URIs of files referenced by the file
 	 */
-	private resolveReferencedFiles(uri: string, span = new Span()): Observable<string> {
-		let observable = this.referencedFiles.get(uri);
+	private resolveReferencedFiles(uri: URL, span = new Span()): Observable<URL> {
+		let observable = this.referencedFiles.get(uri.href);
 		if (observable) {
 			return observable;
 		}
-		const parts = url.parse(uri);
-		if (!parts.pathname) {
-			return Observable.throw(new Error(`Invalid URI ${uri}`));
-		}
 		// TypeScript works with file paths, not URIs
-		const filePath = parts.pathname.split('/').map(decodeURIComponent).join('/');
+		const filePath = uri.pathname.split('/').map(decodeURIComponent).join('/');
```

Comment:
I would want a Windows user to see `C:\Users\felix\myfile.ts` as a `containerName` for a class though, not `/C:/Users/felix/myfile.ts`